packages/element/src/create-interpolate-element.js

```diff
@@ -0,0 +1,235 @@
+ * Internal dependencies
+ */
+import { createElement, Fragment, isValidElement } from './react';
+
+/**
+ * External dependencies
+ */
+import { escapeRegExp, flatMap } from 'lodash';
+
+const getHasPropValue = ( config ) => !! config.value;
+const getHasChildren = ( config ) => !! config.hasChildren;
+
+/**
+ * Transforms a given string value into a collection of html like tags.
+ *
+ * @example
+ *
+ * If you provide `something` as the tagString, this will return:
+ *
+ * ```js
+ * const tags = getTagsFromString( 'something' );
+ * const tagsAre = tags === [
+ *     '<something>',
+ *     '</something>',
+ *     '<something/>',
+ * ];
+ * ```
+ *
+ * @param {string} tagString  Something like `a%1`.
+ *
+ * @return {string[]} The generated tags for the given string.
+ */
+const getTagsFromString = ( tagString ) => {
+	return [
+		`<${ tagString }>`,
+		`</${ tagString }>`,
+		`<${ tagString }/>`,
+	];
+};
+
+/**
+ * Generates and returns the regular expression for the given type.
+ *
+ * @param {Array}		tags          Array of tags to use for the reg ex
+ * @param {string}  searchString  The search string serving as the base for the
+ *                                expression.
+ * @param {string}	type					What type of regEx to return.
+ *
+ * @return {RegExp}  The generated regular expression.
+ */
+const getRegEx = ( tags, searchString, type ) => {
+	let pattern;
+	const [ openTag, closeTag, selfClosingTag ] = tags;
+	switch ( type ) {
+		case 'children':
+			pattern = escapeRegExp( openTag ) + '(.*)' + escapeRegExp( closeTag );
+			break;
+		case 'propValue':
+			pattern = escapeRegExp( searchString );
+			break;
+		default:
+			pattern = escapeRegExp( selfClosingTag );
+	}
+	return new RegExp( pattern );
+};
+
+/**
+ * Generates and returns the regular expression used for splitting a string
+ *
+ * @param {string} searchString  The search string serving as the base for the
+ *                               expression.
+ *
+ * @return {RegExp}  The generated regular expression
+ */
+const getSplitRegEx = ( searchString ) => new RegExp( '(' + escapeRegExp( searchString ) + ')' );
+
+/**
+ * Generates a String.prototype.match value for the incoming arguments.
+ *
+ * @param {string} interpolatedString  The string the match is performed on.
+ * @param {string} searchString				 The string used as the base for the
+ *                                     search
+ * @param {Object} conversionConfig    The configuration object for the
+ *                                     interpolation being performed.
+ *
+ * @return {Array|null}	An array if there is a match or null if not.
+ */
+const getMatchFromString = (
+	interpolatedString,
+	searchString,
+	conversionConfig
+) => {
+	const tags = getTagsFromString( searchString );
+	// first try children reg ex.  If there is a match, then return.
+	const match = interpolatedString.match(
+		getRegEx( tags, searchString, 'children' )
+	);
+	if ( match !== null ) {
+		conversionConfig.hasChildren = true;
```

Comment:
The risk of actual unexpected mutations is probably low, but I wonder if we should deeply clone the conversion map before calling `recursiveCreateElement`. Or carry over the `hasChildren` datum by some other means.

Reply:
Hmm... the risk will primarily be if the conversion map is used in sharing the conversionMap they provide to `createInterpolateElement` with some other logic outside of here.  So ya it's low, but does present potential for a gotcha.  If we change the interface to something like the interface comment discussion, then I can clone the config before mutating it (or look at doing what you suggest with communicating the `hasChildren` type some other way.

Reply:
I decided to leave this as is.  We can see how the experiment plays out but configs are intended to be one use and context specific.  The only place this mutation would be a problem is if consumers share configurations across localized strings which is highly unlikely (and any use-cases where I might see that happen would be in conditional statements where mutation would not matter).